modules/assistant/session.go

```diff
@@ -463,50 +501,97 @@ func (h APIHandler) openChatSession(w http.ResponseWriter, req *http.Request, ps
+	builder.Exclude("message", "attachments", "details", "parameters")
+
+	ctx := orm.NewContext()
+	orm.WithModel(ctx, &common.ChatMessage{})
+
+	res, err := orm.SearchV2(ctx, builder)
+	if err != nil {
+		return nil, err
+	}
+
 	docs := []common.ChatMessage{}
-	err, _ := orm.SearchWithJSONMapper(&docs, &q)
+	err = util.FromJSONBytes(res.Payload.([]byte), &docs)
 	if err != nil {
 		return nil, err
 	}
 	return docs, nil
 }
 
 func (h APIHandler) getChatHistoryBySession(w http.ResponseWriter, req *http.Request, ps httprouter.Params) {
-	q := orm.Query{}
-	q.Conds = orm.And(orm.Eq("session_id", ps.MustGetParameter("session_id")))
-	q.From = h.GetIntOrDefault(req, "from", 0)
-	q.Size = h.GetIntOrDefault(req, "size", 20)
-	q.AddSort("created", orm.ASC)
+	builder := orm.NewQuery()
+	builder.Must(orm.TermQuery("session_id", ps.MustGetParameter("session_id")))
+	builder.From(h.GetIntOrDefault(req, "from", 0))
+	builder.Size(h.GetIntOrDefault(req, "size", 20))
+	builder.SortBy(orm.Sort{Field: "created", SortType: orm.ASC})
 
-	err, res := orm.Search(&common.ChatMessage{}, &q)
+	// Check if full content is requested via query parameter
+	fullContent := h.GetParameterOrDefault(req, "full_content", "false") == "true"
+
+	if !fullContent {
+		// Use projection for lighter queries by default
+		// Include only essential fields for chat message listing
+		builder.Include("id", "created", "updated", "type", "session_id", "from", "to", "assistant_id", "reply_to_message", "up_vote", "down_vote")
+		// Exclude heavy content fields
+		builder.Exclude("message", "attachments", "details", "parameters")
+	}
+
+	ctx := orm.NewContextWithParent(req.Context())
+	orm.WithModel(ctx, &common.ChatMessage{})
+
+	res, err := orm.SearchV2(ctx, builder)
 	if err != nil {
 		h.WriteError(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
 
-	_, err = h.Write(w, res.Raw)
+	_, err = h.Write(w, res.Payload.([]byte))
 	if err != nil {
 		h.Error(w, err)
 	}
 }
 
 var inflightMessages = sync.Map{}
 
+// MessageTask represents an active message processing task
 type MessageTask struct {
-	SessionID string
-	// Deprecated
-	TaskID string
-	// Deprecated
-	WebsocketID string
+	SessionID   string
+	TaskID      string // Deprecated
+	WebsocketID string // Deprecated
 	CancelFunc  func()
+	CreatedAt   time.Time // For TTL-based cleanup
 }
 
+// TaskManager handles cleanup of inflight message tasks
+type TaskManager struct {
+	cleanupInterval time.Duration
+	taskTTL         time.Duration
+	stopChan        chan struct{}
+}
+
+var taskManager *TaskManager
+
 func init() {
+	// Initialize task manager with reasonable defaults
+	taskManager = &TaskManager{
+		cleanupInterval: 5 * time.Minute,  // Run cleanup every 5 minutes
+		taskTTL:         30 * time.Minute, // Tasks expire after 30 minutes
+		stopChan:        make(chan struct{}),
+	}
+
+	// Start background cleanup goroutine
+	go taskManager.startCleanup()
```

Comment:
rather the general `go`, prefer to using task module, as it is with more visibility to user and with more controls